x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -99,19 +111,7 @@ public static List<ValueSourceInfo> sources(
             var fieldContext = new FieldContext(fieldName, fieldData, fieldType);
             var vsType = fieldData.getValuesSourceType();
             var vs = vsType.getField(fieldContext, null);
-
-            if (asUnsupportedSource) {
```

Comment:
This was moved earlier in the method because there is no need for all the checks before it, if we know that the data type of the field is definitely an unsupported one.